GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -45,23 +43,137 @@
 import org.gearvrf.scene_objects.GVRModelSceneObject;
 import org.gearvrf.utility.FileNameUtils;
 import org.gearvrf.utility.Log;
-
 import android.content.Context;
 import android.content.res.AssetManager;
 import android.os.Environment;
 import org.gearvrf.jassimp.AiColor;
+
 /**
- * {@link GVRImporter} provides methods for importing 3D models and making them
+ * {@link GVRAssetLoader} provides methods for importing 3D models and making them
  * available through instances of {@link GVRAssimpImporter}.
  * <p>
  * Supports importing models from an application's resources (both
  * {@code assets} and {@code res/raw}) and from directories on the device's SD
  * card that the application has permission to read.
  */
-final class GVRImporter {
-    private GVRImporter() {
-    }
+public final class GVRAssetLoader implements IEventReceiver {
+    /**
+     * Listens for texture load events and raises the "onAssetLoaded"
+     * event after all textures have been loaded.
+     */
+    static public class AssetEventListener implements IAssetEvents
+    {
+        protected GVRAssetLoader       mImporter = null;
+        protected int               mNumTextures = 0;
+        protected GVRScene          mScene = null;
+        protected GVRSceneObject    mModel = null;
+        protected IAssetEvents      mUserHandler = null;
+        protected String            mFileName;
+        protected String            mErrors;
+        
+        public AssetEventListener(GVRAssetLoader importer, GVRScene scene) {
+            mScene = scene;
+            mImporter = importer;
+            mNumTextures = 0;
+        }
+        public void beginLoadTexture()
+        {
+            ++mNumTextures;
+        }
+        
+        public void beginLoadAsset(String filePath, IAssetEvents userHandler)
+        {
+            mImporter.getEventReceiver().addListener(userHandler);
+            mFileName = filePath;
+            mNumTextures = 0;
```

Comment:
We will eventually support passing in an asset listener so the caller can trap when models and textures are loaded and modify them before adding them to the scene. Since our asset loading API is already so complex and nobody has actually asked for this yet I did not expose the asset listener to the user.
Sometimes you want to do something like flip all the textures in the Y direction for a set of assets. Exposing the asset listener feature would allow it. I would like to simplify the asset loading first before we add new stuff.